pkg/workflows/management/post_cluster_upgrade.go

```diff
@@ -28,6 +28,10 @@ func (s *postClusterUpgrade) Run(ctx context.Context, commandContext *task.Comma
 		logger.Info(fmt.Sprintf("management cluster CAPI backup file not found: %v", err))
 	}
 
+	if commandContext.CurrentClusterSpec.Cluster.Spec.RegistryMirrorConfiguration != commandContext.ClusterSpec.Cluster.Spec.RegistryMirrorConfiguration {
```

Comment:
I'm not sure if this will work very well since this field is a pointer

Regardless, there is an `Equal` method, I would vote to use that